tests/run-make/parallel-rustc-no-overwrite/rmake.rs

```diff
@@ -0,0 +1,22 @@
+// When two instances of rustc are invoked in parallel, they
+// can conflict on their temporary files and overwrite each others',
+// leading to unsuccessful compilation. The -Z temps-dir flag adds
+// separate designated directories for each rustc invocation, preventing
+// conflicts. This test uses this flag and checks for successful compilation.
+// See https://github.com/rust-lang/rust/pull/83846
+
+use run_make_support::{fs_wrapper, rustc};
```

Comment:
Looks like you committed this one by accident? :)

Reply:
By "this one", you mean the test? No, this is me renaming "issue-10971-temps-dir" because I thought this name was clearer.

Reply:
Ah, sorry, I'm stupid, didn't realize that :)

You don't need the second thread, we can run the second rustc in the main thread in `main` directly.
Also I would consider using a [barrier](https://doc.rust-lang.org/std/sync/struct.Barrier.html) to increase the chance that these two things will really run at the same time. In theory rustc could be finished by the time the second thread is even started.

Reply:
Hey, that's a really cool feature! I've never seen it before. I might not get the implementation right on the first try, but I just gave it a go.

Reply:
Yeah, that looks great :) That's exactly what I meant.